### PR TITLE
Major update to CI workflows and dependencies

### PR DESCRIPTION
**Pull request by gaby**
## Change Summary
- This PR is a continuation of #285 #286 
- Updated all dependencies to their latest version
- Updated all dev dependencies to their latest version
- Removed support for python 3.6
- Removed support for python 3.7 (EOL - June 2023)
- Refresh the poetry.lock file
- Added `macos-latest` to test matrix
- Used [flynt](https://github.com/ikamensh/flynt) to update as many strings to `f-strings`
- Migrated deprecated `[tool.poetry.dev-dependencies]` to a dependency group

## TODO
- @tfeldmann Would need to update the package version and do a new release

## Checklist
- [x] Tests for the changes exist and pass on CI
- [x] My PR is ready to review

## Tests Passing

image

### CODE REVIEW

**Comment by gaby** [comment]
This PR closes #280 #241 #240 #247 #248

**Comment by gaby** [comment]
@tfeldmann When you get a chance, can you take a look at this PR.

**Comment by tfeldmann** [comment]
Thank you for this PR. Looks really good. I think poetry manages the classifiers automatically, so adding them to the pyproject.toml should not be needed? I'll try this one locally and then merge it.

File: pyproject.toml [hunk 1]
```diff
@@ -39,18 +37,18 @@ classifiers = [
 organize = "organize.cli:cli"
 
 [tool.poetry.dependencies]
-python = "^3.6.2"
+python = ">=3.8,<3.12"
```
[the remark this answers is not in the document]
Comment (gaby) [reply pyproject.toml:40]: Py 3.7 is EOL since June, 2023

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -28,12 +28,14 @@ jobs:
...
         run: |
-          python3 -m pip install --upgrade pip setuptools
-          python3 -m pip install poetry
+          python3 -m pip install --upgrade pip setuptools poetry
           poetry run python -m pip install -U pip setuptools
-          poetry install -E textract
+
+      - name: Install dependencies
+        run: |
+          poetry install --with dev -E textract || poetry install --with dev -E textract || echo "No dev group"
```
[the remark this answers is not in the document]
Comment (gaby) [reply .github/workflows/tests.yml:38]: That's the fix suggested by Poetry, it's either that or rolling back to an older version of Poetry. The weird part is that it only affects CI/CD, not local stuff

[the remark this answers is not in the document]
Comment (gaby) [reply .github/workflows/tests.yml:38]: @tfeldmann The solution came from here: https://github.com/python-poetry/poetry/issues/7611#issuecomment-1510984401

**Comment by gaby** [comment]
> Thank you for this PR. Looks really good. I think poetry manages the classifiers automatically, so adding them to the pyproject.toml should not be needed? I'll try this one locally and then merge it.

They are optional according to the docs: https://python-poetry.org/docs/pyproject/#classifiers

**Comment by tfeldmann** [comment]
Thank you!

**Pull request merged**